pkg/storage/tpr/obj_state.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tpr
+
+import (
+	"errors"
+	"fmt"
+
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/storage"
+)
+
+var (
+	errResourceVersionCannotBeSet = errors.New("resource version cannot be set on objects")
+)
+
+type errCouldntGetResourceVersion struct {
+	err error
+}
+
+func (e errCouldntGetResourceVersion) Error() string {
+	return fmt.Sprintf("couldn't get resource version (%s)", e.err)
+}
+
+type objState struct {
```

Comment:
It's certainly useful for people writing TPR `storage.Interface` implementations. I think that this whole suite of TPR-based `storage.Interface` stuff should be pushed upstream, and soon (possibly to `k8s.io/apiserver`)